feat: new translation languages

kimhw0630 opened the pull request:
Add translations from TEW team and supporting Japanese, German, and Chinese except for two features: OPF and User Profile.

Technical details:
- in `translations.ts` barrel files now we export each langauge in a separate public API members (to allow for treeshaking unused langauges), e.g. `translationsEn`, `translationsDe`,... - in each library
- deprecate the the old wrapper consts e.g. `translations` (and moved those deprecated tokens to `public_api.ts` files from `translations.ts` files in each library)
- add a new function `extractTranslationChunksConfig()` in `@spartacus/core` and use it in every library for creating chunks config in their `translation.ts` files. It extracts the chunks config from the translations assets automatically. So no more need to type chunks config manually. Aa consequence a few human errors in the chunks configuration were fixed  implicitly by the new util `extractTranslationChunksConfig()` (you won't find it in the code, but only in runtime):
    - added a missing array at `cpqquoteTranslationChunksConfig.cpqQuote`
    - removed unnecessary array at `cpqquoteTranslationChunksConfig.cpqQuoteHeading`
    - removed values from the array `translationChunksConfig.myAccount` - the keys that have been moved to other files long time ago, but the leftover values in the chunks config remained (`"orderDetails",  "orderHistory",  "AccountOrderHistoryTabContainer",  "returnRequestList",  "returnRequest"`)
    - removed the obsolete value `"CMSTabParagraphContainer"` from the array at `translationChunksConfig.product` - such a key no longer exists in translations since some time, but a leftover remained in the chunks config
    - removed the obsolete value `"loginRegister"` from the array at `translationChunksConfig.user` - such a key no longer exists in translations since some time, but a leftover remained in the chunks config
- fix our installation schematics, to use the new recommended syntax in the scaffolded features e.g.:
   ```diff
   - resources: asmTranslations,
   + resources: { en: asmTranslationsEn }
   ```
- use the above new approach in all feature modules in the storefrontapp
- renamed files to match their chunk names: `recentSearches.json` -> `cdcRecentSearches.json`, `trendingSearches.json` -> `cdsTrendingSearches.json`

fixes https://jira.tools.sap/browse/CXSPA-7172

hakwookim committed:
translation configuration file added

hakwookim committed:
updated targetFolderPath

hakwookim committed:
updated translation_v2 suggest from TEW team

service-tip-git committed:
Translation Delivery (#19524)

hakwookim committed:
Merge remote-tracking branch 'origin/develop' into epic/tew-translation

hakwookim committed:
changed ts to json resources from user profile

hakwookim committed:
setting 4 language except user profile lib

hakwookim committed:
Merge remote-tracking branch 'origin/develop' into epic/tew-translation

hakwookim committed:
Trigger Build

hakwookim committed:
format files using npm run prettier:fix

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

pawelfras commented on feature-libs/asm/assets/translations/ja/asm.json:
```diff
@@ -0,0 +1,175 @@
+    "createCustomerSuccessfullyAlert": "The customer account has been created and the customer session has started.",
+    "saveInactiveCartAlertInfo": "The identified cart is an inactive cart. To take further actions on this cart, save it first.",
+    "activeCartAlertInfo": "The identified cart is an active cart.",
+    "startCustomerEmulationAlertInfo": "Customer emulation has started. Any actions you do will reflect the effects on the customer account.",
```
`major:` There are many values across all new translation files that are not translated to the particular language. Do you perhaps know if there is any official process to inform TEW about these gotchas?

pawelfras reviewed (commented):
Unfortunately, translation files are not tree-shakeable and they be a part of the main bundle. Right now, main.js of storefrontapp's prod build has grown by 300kb, and it will grow each time we add new translation files/languages. I'm checking if we can load these files dynamically to reduce the initial load. Probably something I'd like to discuss with @Platonn.

kimhw0630 committed:
Merge branch 'develop-next-major' into epic/tew-translation

service-tip-git committed:
[INTERNAL] Translation delivery: commit by LX Lab

Change-Id: I76946206a37bd45386109c314c8636c708cbc738

Platonn committed:
Merge branch 'develop-next-major' of github.com:SAP/spartacus into epic/tew-translation

github-actions[bot] committed:
Add license header

Platonn committed:
chore(license header): remove obsolete line with 2024, but leave only 2025

Platonn committed:
feat: allow for tree-shaking unused langauges translations in custom apps (#19799)

- add a lot of missing `index.ts` files to re-export JSON translations in TS for new langauges
- update `translations.ts` barrel files, to export each langauge in a separate public API members (to allow for treeshaking unused langauges). Note: it preservers the old feature prefix (e.g. `asmTranslations` -> `asmTranslationsEn`, `asmTranslationsDe`, ...)
- deprecate the the old wrapper consts e.g. `asmTranslations` (and moved those deprecated tokens to `public_api.ts` files from `translations.ts` files)
- remove from the old wrapper consts e.g. `asmTranslations` langauges other than `en` (in other words: revert the change made previously on the branch [epic/tew-translation](https://github.com/SAP/spartacus/tree/epic/tew-translation) which added 3 other langauges into those wrapper objects)
- add a new function `extractTranslationChunksConfig()` in `@spartacus/core` and use it in every library for creating chunks config in their `translation.ts` files. It extracts the chunks config from the translations assets automatically. So no more need to type chunks config manually.
- fix our installation schematics, to use the new recommended syntax in the scaffolded features e.g.:
   ```diff
   - resources: asmTranslations,
   + resources: { en: asmTranslationsEn }
   ```
   
By the way:
  - noticed a broken chunks mapping for the cpqQuote feature. now should be fixed since using function `extractTranslationChunksConfig()`
  - noticed a typo in the existing old chunk name `myAccountV2NotifiationPerference`, HOWEVER I didn't fix it in this PR - it should be done in a separate PR
  - noticed yet another typo in existing file name `my-account-v2-notification-perference.json` (btw. it's a different typo than the above chunk name), HOWEVER I didnt fix it in this PR
  - noticed some i18n JSON files don't use camelCase, but kebab-case, HOWEVER I didn't fix it in this PR - it should be done in a separate PR
  
The reason I didn't want to fix those things by the way is:
- it might affect the workflow of the TEW team, and I'd need to first double check with them
- it's not related to this PR's main purpose (three-shaking)

TODO
- [x] fix unit tests and Jest snapshots in schematics
- [x] use it this way also in our storefrontapp
- [ ] write doc input for the migration docs (to use specific langauges)


related to https://jira.tools.sap/browse/CXSPA-9131

**QA steps**
1. checkout to this branch
2. build libs
3. publish to verdaccio
4. create a new app `npx @angular/cli@17 new test-app --standalone=false --ssr=false`
6. in the new app, install sparatcus from verdaccio `ng add @spartacus/schematics@2211.32. 0-1 --baseUrl="https://40.76.109. 9:9002" --ssr` AND PLEASE SELECT ALL possible features and integrations in the interactive prompt
7. In VsCode search everywhere (Cmd+Shift+F) for the phrase `resources:` and verify that everywhere in every feature module we use the new approach with explicit EN translations
8. In some of those modules try to change the approach to old `translations` wrapper object and verify it's `@deprecated` (crossed out in VSCode)
9. in spartacus-configuration.module.ts add also German translations for the main translations:
   ```ts
   i18n: {
        resources: { en: translationsEn, de: translationsDe },
   ```
10. in spartacus-configuration.module.ts add this config (it's needed for later to run `http-server`)
   ```ts
   provideConfig(<SiteContextConfig>{
      context: {
        baseSite: ['electronics-spa', 'apparel-uk-spa'],
        currency: ['USD', 'GBP'],
        language: ['en', 'de'],
      },
    }),
   ```
11. `ng build` (or for interactive mode run `npm run watch`). Don't try `ng serve --configuration=production` - it doesn't have so well tree shaking as `ng build`
12. `npx http-server dist/test-app/browser` (or for interactive mode run in another terminal `npx nodemon --watch "./dist/test-app/browser" --exec "npx http-server './dist/$test-app/browser'" --ext "*"`
13. In Chrome open http://127.0.0.1:8080/ (8080! not 4200 port!)
14. In devtools Sources tab do the search (Cmd+F)
15. paste this search phrase `ariaLabelSuggestions` and verify that in the loaded `main.js` file there are only 2 langauges (EN and DE which you have enabled in steps above) with translations for this key
    <img width="512" alt="image" src="https://github.com/user-attachments/assets/f6be33e1-7cc7-4e2a-bde3-ecdf4a19db81" />

hakwookim committed:
executed prettier

hakwookim committed:
fixed e2e

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

hakwookim committed:
Trigger Build

Platonn committed:
fix: bring back the old chunks config for the CDS with the `cds` prefix (#19888)

**Problem:**
When lately double-checking in runtime whether the translation chunks configs for all features are the same Before the PR https://github.com/SAP/spartacus/pull/19799 and After it, I noticed a small difference. The chunks config for the CDS feature unintentionally lost their `cds` prefix. 

Note:  For consistency also renamed the relevant JSON filenames to match with the desired `cds`-prefixed chunk names.


---
**QA steps**
1. add this APP_INTIALIZER to the providers array in our app.module:
```ts
import { cdsTranslationChunksConfig } from '@spartacus/cds/assets';
/*...*/

    {
      provide: APP_INITIALIZER,
      useFactory: () => {
        console.log({
          cdsTranslationChunksConfig,
        });
      },
      multi: true,
    },
 ```

2. Run and open the storefront, and check in runtime in the console the printed value contains:
EXPECTED (with `cds`-prefixed keys):
```
"cdsTranslationChunksConfig": {
    "cdsRecentSearches": ["cdsRecentSearches"],
    "cdsTrendingSearches": ["cdsTrendingSearches"]
  },
```

WRONG (it was before this commit):
```
  "cdsTranslationChunksConfig": {
    "recentSearches": ["cdsRecentSearches"],
    "trendingSearches": ["cdsTrendingSearches"]
  },
```

hakwookim committed:
Trigger Build

hakwookim committed:
Merge remote-tracking branch 'origin/develop-next-major' into epic/tew-translation

hakwookim committed:
assets version updated
